core/src/main/scala/org/apache/spark/metrics/sink/PrometheusPushGateWay.scala

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.metrics.sink
+
+import java.util
+import java.util.Properties
+
+import com.codahale.metrics.MetricRegistry
+import io.prometheus.client.{CollectorRegistry, Gauge}
+import io.prometheus.client.exporter.PushGateway
+import org.apache.commons.lang3.StringUtils
+
+import org.apache.spark.SecurityManager
+
+/**
+ * PrometheusPushGateWay that exports Metric Metrics via Prometheus PushGateway.
+ */
+private[spark] class PrometheusPushGateWay(
```

Comment:
Why not having this as a third party library?

Reply:
> Could you explain the difference from #19775?

hi @dongjoon-hyun Thank you very much, There is no difference in functionality. I don't see it in the spark source code, so I implemented it myself. #19775 is better written. I will close this PR.